File: sgilt-front/app/data/ficheia/dto/FicheIaGenerationDto.ts
/**
 * DTO — contrat API pour la génération IA de fiche prestataire
 */
import type { Testimonial } from '~/data/prestataire/domain/Testimonial'
import type { DetailItem } from '~/data/prestataire/domain/DetailItem'
import type { FaqItem } from '~/data/prestataire/domain/FaqItem'

export interface FicheIaGenerationContentDto {
  shortDescription: string
  baseline: string
  offerings: string[]
  identity: { quote: string; bio: string }
  testimonials: Testimonial[]
  details: DetailItem[]
  faq: FaqItem[]
  budget: string
}

export interface FicheIaGenerationResultDto {
  result: FicheIaGenerationContentDto | null
  triesLeft: number
  lastGenerationDateTime: string | null
}

File: sgilt-core/src/main/java/net/franzka/sgilt/core/ficheia/dto/FicheIaGenerationDto.java
package net.franzka.sgilt.core.ficheia.dto;

import net.franzka.sgilt.core.prestataire.dto.DetailDto;
import net.franzka.sgilt.core.prestataire.dto.FaqItemDto;
import net.franzka.sgilt.core.prestataire.dto.IdentityDto;
import net.franzka.sgilt.core.prestataire.dto.TestimonialDto;

import java.util.List;

/**
 * DTO du contenu de fiche prestataire généré par IA, avec des champs alignés sur ceux de
 * l'entité {@code Prestataire} pour permettre un mapping direct dans les briques ultérieures.
 */
public record FicheIaGenerationDto(
        String shortDescription,
        String baseline,
        List<String> offerings,
        IdentityDto identity,
        List<TestimonialDto> testimonials,
        List<DetailDto> details,
        List<FaqItemDto> faq,
        String budget
) {}
